lib/package-manager/get-executable.js

```diff
@@ -1,15 +1,20 @@
-'use strict';
+import { dirname } from 'path';
+import { promisify } from 'util';
+import { fileURLToPath } from 'url';
 
-const { promisify } = require('util');
+import npmWhichLib from 'npm-which';
+import whichLib from 'which';
 
-const npmWhich = promisify(require('npm-which')(__dirname));
-const which = promisify(require('which'));
+const npmWhich = promisify(
+  npmWhichLib(dirname(fileURLToPath(import.meta.url)))
```

Comment:
This is ugly. Can we find a better way?

Reply:
This seems like feedback to bring to modules team about better methods for dirname / filename perhaps?